src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php

```diff
@@ -220,17 +220,15 @@ private function getDocBlockFromProperty(string $class, string $property): ?DocB
             return null;
         }
 
-        try {
-            $reflector = $reflectionProperty->getDeclaringClass();
-
-            foreach ($reflector->getTraits() as $trait) {
-                if ($trait->hasProperty($property)) {
-                    $reflector = $trait;
+        $reflector = $reflectionProperty->getDeclaringClass();
 
-                    break;
-                }
+        foreach ($reflector->getTraits() as $trait) {
+            if ($trait->hasProperty($property)) {
+                return $this->getDocBlockFromProperty($trait->getName(), $property);
             }
+        }
 
+        try {
```

Comment:
Note: I moved the login out the `try/catch` block because as far as I know, this statement is for the `$this->docBlockFactory->create` line below, not for the recursive resolution.

Please tell me if I'm wrong. :+1: